### PR TITLE
Creating test module

### PR DESCRIPTION
**Pull request by katlazarova**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
         "vlucas/phpdotenv": "^2.4",
         "webflo/drupal-finder": "^1.0.0",
         "webmozart/path-util": "^2.3",
-        "zaporylie/composer-drupal-optimizations": "^1.0"
+        "zaporylie/composer-drupal-optimizations": "^1.0",
+        "ext-json": "*"
```
Comment (ElXyfer) [inline composer.json]: Remove this and use this instead: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Serialization%21Json.php/class/Json/8.2.x

File: config/sync/core.extension.yml [hunk 2]
```diff
@@ -45,6 +45,8 @@ module:
   page_cache: 0
   path: 0
   pineapple_general: 0
+  pineapple_hello_world: 0
```
Comment (ElXyfer) [inline config/sync/core.extension.yml]: Uninstall this and delete it.

File: web/modules/custom/pineapple_helper/pineapple_helper.module [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      ),
+      '#cache' => [
+        'contexts' => [
+          'route.name',
+        ],
+      ],
+    );
+
+    // Create a form element of type link, which is a title suffix.
+    $variables['title_suffix']['help_form'] = array(
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/pineapple_helper.module]: Consider naming this element something else

File: web/modules/custom/pineapple_helper/pineapple_helper.routing.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  requirements:
+    _permission: 'access administration pages'
+
+pineapple_helper.main:
+  path: '/admin/pineapple-helper/raise-issue'
+  defaults:
+    _controller: '\Drupal\pineapple_helper\Controller\RaiseIssueDialogController::build'
+    _title: 'Raise an Issue'
+  requirements:
+    _permission: 'access administration pages'
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/pineapple_helper.routing.yml]: Remove this as it is duplicated.

File: web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+pineapple_helper.main:
+  title: Report an Issue
+  description: 'Report an issue'
+  route_name: pineapple_helper.main
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml]: Use `pineapple_helper.raise_issue_dialog` instead

File: web/modules/custom/pineapple_helper/pineapple_helper.module [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      '#attributes' => array(
+        'class' => [
+          'pineapple-helper__spacer'
+        ],
+      ),
+      '#cache' => [
+        'contexts' => [
+          'route.name',
+        ],
+      ],
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/pineapple_helper.module]: Ready into drupal caching and include comment for this.

File: web/modules/custom/pineapple_helper/pineapple_helper.module [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      '#title' => t('Have an issue? Drop us a message'),
+      '#url' => Url::fromRoute('pineapple_helper.raise_issue_dialog'),
+      '#attributes' => array(
+        'class' => [
+          'pineapple-helper-link__button',
+          'button',
+          'button--action',
+          'button--small',
+          'use-ajax'
+        ],
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/pineapple_helper.module]: Fix button margin styling issue.

File: web/modules/custom/pineapple_helper/templates/raise-issue-dialog.html.twig [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<!-- Add you custom twig html here -->
+<p>
+  <strong>Report an Issue</strong>
+</p>
+<a href="/contact" class="button button--primary" target="_blank">Report an issue</a>
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/templates/raise-issue-dialog.html.twig]: Add more content which is helpful for the user.

File: web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+  weight: 10
+  parent: system.admin
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml]: Add options here that include the ajax class added on the button and also include the data attribute added on that button. Examples of formatting should be found on other modules or in the "array"

File: composer.json (hunk 1 above)
Comment (katlazarova) [reply composer.json]: Done

File: config/sync/core.extension.yml (hunk 2 above)
Comment (katlazarova) [reply config/sync/core.extension.yml]: Done

File: web/modules/custom/pineapple_helper/pineapple_helper.module (hunk 3 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/pineapple_helper.module]: Done

File: web/modules/custom/pineapple_helper/pineapple_helper.routing.yml (hunk 4 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/pineapple_helper.routing.yml]: Done

File: web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml (hunk 5 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml]: Done

File: web/modules/custom/pineapple_helper/pineapple_helper.module (hunk 6 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/pineapple_helper.module]: done.

File: web/modules/custom/pineapple_helper/pineapple_helper.module (hunk 7 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/pineapple_helper.module]: Done

File: web/modules/custom/pineapple_helper/templates/raise-issue-dialog.html.twig (hunk 8 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/templates/raise-issue-dialog.html.twig]: Done

File: web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml (hunk 9 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_helper/pineapple_helper.links.menu.yml]: done

**Pull request merged**